### ISSUE TITLE
ListGroup and ListItem should be <ul> and <li> accordingly

### ISSUE THREAD

**Issue by mtscout6** [react-bootstrap/react-bootstrap#460]
Currently it's a combination of divs and spans.

**Comment by mtscout6** [comment]
Bootstrap supports [both](http://getbootstrap.com/components/#list-group). The two options should not be mixed and matched. Which I believe is partly the reason for the `spans`. Consider the following example:

```
<ListGroup>
  <ListGroupItem>Non Link</ListGroupItem>
  <ListGroupItem href='#'>Link</ListGroupItem>
  <ListGroupItem onClick={function() { alert('clicked'); }}>Link</ListGroupItem>
</ListGroup>
```

In this case what do we do? The first item suggests `ul`s with `li`s. Yet the second two suggest `div`s with `a`s. Although the last should really be a button for accessibility and proper semantic html purposes. For clarity sake I tried the following raw html:

```
<div>
  <ul class="list-group">
    <li class="list-group-item">Dapibus ac facilisis in</li>
    <li class="list-group-item">Cras sit amet nibh libero</li>
    <li class="list-group-item">Porta ac consectetur ac</li>
    <li class="list-group-item">
      <a href='#'>Vestibulum at eros</a>
    </li>
  </ul>
  <div class="list-group">
    <a href="#" class="list-group-item">Dapibus ac facilisis in</a>
    <a href="#" class="list-group-item">Cras sit amet nibh libero</a>
    <a href="#" class="list-group-item">Porta ac consectetur ac</a>
    <span class="list-group-item">Porta ac consectetur ac</span>
    <button class="list-group-item">Vestibulum at eros</button>
  </div>
</div>
```

Which produces:

![](http://cl.ly/image/1a3D1N3P2l11/Image%202015-04-01%20at%209.11.42%20AM.png)

Issues to solve:

- [ ] Check all children for `href`s.
- [ ] Still allow for the `span` when rendering a parent `div` and the item is not a link.
- [ ] Use `<button>`s instead of anchors for items with `onClick` -- Blocked by: twbs/bootstrap#16215

This action plan may change pending the outcome of twbs/bootstrap#16204

**Comment by mtscout6** [comment]
Woohoo Bootstrap 3.3.5 now supports buttons!

**Comment by taion** [comment]
Superseded by #1181.
